nbviewer/providers/local/handlers.py

```diff
@@ -172,26 +202,21 @@ def show_dir(self,  abspath,  path):
 
         for f in contents:
             absf = os.path.join(abspath, f)
+
+            if not self.can_show(absf):
```

Comment:
Note: the cost of the factorisation is that we now do two ``os.stat`` calls on each file/folder, not 1.